static/app/views/dashboards/widgets/areaChartWidget/areaChartWidgetVisualization.tsx

```diff
@@ -0,0 +1,216 @@
+    saveOnZoom: true,
+  });
+
+  // TODO: There's a TypeScript indexing error here. This _could_ in theory be
+  // `undefined`. We need to guard against this in the parent component, and
+  // show an error.
+  const firstSeries = props.timeseries[0];
+
+  // TODO: Raise error if attempting to plot series of different types or units
+  const firstSeriesField = firstSeries?.field;
+  const type = meta?.fields?.[firstSeriesField] ?? 'number';
+  const unit = meta?.units?.[firstSeriesField] ?? undefined;
+
+  const formatter: TooltipFormatterCallback<TopLevelFormatterParams> = (
+    params,
+    asyncTicket
+  ) => {
+    // Only show the tooltip of the current chart. Otherwise, all tooltips
+    // in the chart group appear.
+    if (!isChartHovered(chartRef?.current)) {
+      return '';
+    }
+
+    let deDupedParams = params;
+
+    if (Array.isArray(params)) {
+      // We split each series into a complete and incomplete series, and they
+      // have the same name. The two series overlap at one point on the chart,
+      // to create a continuous line. This code prevents both series from
+      // showing up on the tooltip
+      const uniqueSeries = new Set<string>();
+
+      deDupedParams = params.filter(param => {
+        // Filter null values from tooltip
+        if (param.value[1] === null) {
+          return false;
+        }
+
+        if (uniqueSeries.has(param.seriesName)) {
+          return false;
+        }
+
+        uniqueSeries.add(param.seriesName);
+        return true;
+      });
+    }
+
+    return getFormatter({
+      isGroupedByDate: true,
+      showTimeInTooltip: true,
+      valueFormatter: value => {
+        return formatTooltipValue(value, type, unit);
+      },
+      truncate: true,
+      utc: utc ?? false,
+    })(deDupedParams, asyncTicket);
+  };
+
+  let visibleSeriesCount = props.timeseries.length;
+  if (releaseSeries) {
+    visibleSeriesCount += 1;
+  }
+
+  const showLegend = visibleSeriesCount > 1;
+
+  return (
+    <BaseChart
+      ref={e => {
+        chartRef.current = e;
+
+        if (e?.getEchartsInstance) {
+          registerWithWidgetSyncContext(e.getEchartsInstance());
+        }
+      }}
+      autoHeightResize
+      series={[
+        ...props.timeseries.map(timeserie => {
+          return AreaSeries({
+            name: timeserie.field,
+            color: timeserie.color,
+            stack: 'area',
+            animation: false,
+            areaStyle: {
+              color: timeserie.color,
+              opacity: 1.0,
+            },
+            data: timeserie.data.map(datum => {
+              return [datum.timestamp, datum.value];
+            }),
+          });
+        }),
+        releaseSeries &&
+          LineSeries({
+            ...releaseSeries,
+            name: releaseSeries.seriesName,
+            data: [],
+          }),
+      ].filter(defined)}
+      grid={{
+        left: 0,
+        top: showLegend ? 25 : 10,
```

Comment:
These look like values that can be turned into constants and shared with line charts

Reply:
Yep! I'll work on that soon, I think I might declare a `TimeseriesChart` and `AreaChart` and `LineChart` will use it under-the-hood. It just depends on how much code is shared